core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -422,15 +422,24 @@ public Path getWorkingDirectory() {
   @SuppressFBWarnings("ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD")
   @Override
   public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throws IOException {
-    Preconditions.checkNotNull(uri.getHost(), PreconditionMessage.URI_HOST_NULL);
-    Preconditions.checkNotNull(uri.getPort(), PreconditionMessage.URI_PORT_NULL);
+    if (!Configuration.getBoolean(PropertyKey.ZOOKEEPER_ENABLED)) {
```

Comment:
can you add a comment that explains why is it okay to have empty host / port when using Zookeeper? thanks